s4/clarity/_internal/factory.py

```diff
@@ -34,23 +34,27 @@ class ElementFactory(object):
     def _strip_params(string):
         return ElementFactory._params_re.sub('', string)
 
-    def __init__(self, lims, element_class):
-        """
-        Creates a new factory to provide interface between Clarity LIMS and the client software.
-        The provided `element_class` will be used to convert the XML records into Python objects
-        and back. There are several class attributes that can be set to provide an interface.
-
-        NAME_ATTRIBUTE: The XML attribute that contains the name of this type of element. Default 'name'.
-
-        REQUEST_PATH: If the class has the REQUEST_PATH attribute then it will
-        be used to override the default path, which is the plural of the element name.
-        It is used in relation to the root URI of the Clarity API.
-
-        BATCH_FLAGS: Batch Flags use the BitField enumeration to create a bit field that is used
-        to determine what batch services are provided for this data type. It defaults to None.
-
-        :type lims: LIMS
-        :type element_class: classobj
+    def __init__(self,
+                 lims,          # type: s4.clarity.LIMS
+                 element_class  # type: Type[ClarityElement]
```

Comment:
I like this, I never did find a way to do the # type: comments for constructors that made sense.